### PR TITLE
Implemented GraphQL

### PR DESCRIPTION
**Pull request by 2AStudios**
Implemented GraphQL.

### CODE REVIEW

**Comment by cswenberg** [comment]
also maybe clean up the tense and "mostly working DataModels" for your commits.  @ningning621 idk how nitpicky yall are for commits but we should try to keep the history as clean as possible.

File: app/src/main/java/com/cornellappdev/android/eatery/network/NetworkUtilities.java [hunk 1]
```diff
@@ -32,4 +49,29 @@ public static String getJSON() {
...
+        eateryCall.enqueue(new ApolloCall.Callback<AllEateriesQuery.Data>() {
+            @Override
+            public void onResponse(@NotNull Response<AllEateriesQuery.Data> response) {
+                eateries = response.data().eateries();
+                EATERIES_LOADED = true;
+            }
+
+            @Override
+            public void onFailure(@NotNull ApolloException e) {
+                Log.d("Network Response", e.getMessage());
```
[the remark this answers is not in the document]
Comment (ningning621) [reply app/src/main/java/com/cornellappdev/android/eatery/network/NetworkUtilities.java]: Also we try to avoid using Log in our master code....

**Comment by ningning621** [comment]
> also maybe clean up the tense and "mostly working DataModels" for your commits. @ningning621 idk how nitpicky yall are for commits but we should try to keep the history as clean as possible.

I'm personally not super picky about the commit messages, but backend is the master of git so we'll follow your standards :)

File: app/build.gradle [hunk 2]
```diff
@@ -31,15 +30,17 @@ dependencies {
...
     implementation 'com.squareup.picasso:picasso:2.71828'
     implementation 'com.facebook.fresco:fresco:1.9.0'
     implementation 'com.android.support.constraint:constraint-layout:1.0.2'
+    implementation 'com.apollographql.apollo:apollo-runtime:1.0.0-alpha5'
+    implementation 'com.apollographql.apollo:apollo-android-support:1.0.0-alpha5'
     testImplementation 'junit:junit:4.12'
     androidTestImplementation 'com.android.support.test:runner:1.0.1'
     androidTestImplementation 'com.android.support.test.espresso:espresso-core:3.0.1'
-    compile 'net.opacapp:multiline-collapsingtoolbar:1.6.0'
+    implementation 'net.opacapp:multiline-collapsingtoolbar:1.6.0'
```
Comment (young-k) [inline app/build.gradle]: isn't it convention to have this section of the file alphabetized?

File: app/src/main/java/com/cornellappdev/android/eatery/MainActivity.java [hunk 3]
```diff
@@ -66,11 +72,18 @@
...
+	public static ApolloClient apolloClient;
+	public static final boolean JSON_FALLBACK = false;
+
 	@Override
 	protected void onCreate(Bundle savedInstanceState) {
 		super.onCreate(savedInstanceState);
 		setTitle("Eatery");
 		setContentView(R.layout.activity_main);
+
+		//Preload Eateries
```
Comment (young-k) [inline app/src/main/java/com/cornellappdev/android/eatery/MainActivity.java]: space after `//`

File: app/src/main/java/com/cornellappdev/android/eatery/MainActivity.java [hunk 4]
```diff
@@ -66,11 +72,18 @@
...
+        NetworkUtilities.getEateries();
```
Comment (young-k) [inline app/src/main/java/com/cornellappdev/android/eatery/MainActivity.java]: this isn't formatted properly, can you check your indentation settings & make sure they sync with what @ningning621 has?

File: app/src/main/java/com/cornellappdev/android/eatery/MainActivity.java [hunk 5]
```diff
@@ -369,9 +382,21 @@ public boolean onQueryTextChange(String newText) {
...
+                    while (!NetworkUtilities.EATERIES_LOADED) {
+                        try {
+                            Thread.sleep(10);
+                        } catch (InterruptedException e) {
+                            e.printStackTrace();
+                        }
+                    }
+                    List<AllEateriesQuery.Eatery> eateries = NetworkUtilities.getEateries();
+                    cafeList = JsonUtilities.parseEateries(eateries, getApplicationContext());
+                }
```
Comment (young-k) [inline app/src/main/java/com/cornellappdev/android/eatery/MainActivity.java]: this indentation also seems off, between this line & the next there's a huge difference

File: app/src/main/java/com/cornellappdev/android/eatery/model/CafeModel.java [hunk 6]
```diff
@@ -134,6 +142,51 @@ public void setHours(LocalDate date, List<Interval> hours) {
...
+		DateTimeFormatter dateFormatter = new DateTimeFormatterBuilder()
+				.parseCaseInsensitive()
+				.appendPattern("yyyy-MM-dd")
+				.toFormatter();
+
+		for(AllEateriesQuery.OperatingHour operatingHour : cafe.operatingHours()){
+			List<LocalDate> localDates = new ArrayList<>();
+			LocalDate localDate = LocalDate.parse(operatingHour.date(),dateFormatter);
+			localDates.add(localDate);
+			for(AllEateriesQuery.Event event : operatingHour.events()){
```
Comment (young-k) [inline app/src/main/java/com/cornellappdev/android/eatery/model/CafeModel.java]: space after `for` here and on line `158`

File: app/src/main/java/com/cornellappdev/android/eatery/model/CafeModel.java [hunk 7]
```diff
@@ -134,6 +142,51 @@ public void setHours(LocalDate date, List<Interval> hours) {
...
+				List<Interval> dailyHours = new ArrayList<>();
+				for(AllEateriesQuery.Menu menu : event.menu()){
+					for(AllEateriesQuery.Item item : menu.items()){
+						if(!cafeItems.contains(item.item())) cafeItems.add(item.item());
```
Comment (young-k) [inline app/src/main/java/com/cornellappdev/android/eatery/model/CafeModel.java]: always have a space after `if`, so `if (`

File: app/src/main/java/com/cornellappdev/android/eatery/model/CafeModel.java [hunk 8]
```diff
@@ -134,6 +142,51 @@ public void setHours(LocalDate date, List<Interval> hours) {
...
+				.toFormatter();
+
+		for(AllEateriesQuery.OperatingHour operatingHour : cafe.operatingHours()){
+			List<LocalDate> localDates = new ArrayList<>();
+			LocalDate localDate = LocalDate.parse(operatingHour.date(),dateFormatter);
+			localDates.add(localDate);
+			for(AllEateriesQuery.Event event : operatingHour.events()){
+				List<Interval> dailyHours = new ArrayList<>();
+				for(AllEateriesQuery.Menu menu : event.menu()){
+					for(AllEateriesQuery.Item item : menu.items()){
```
Comment (young-k) [inline app/src/main/java/com/cornellappdev/android/eatery/model/CafeModel.java]: space after `)`, so `) {`, can you set up a linter so we can avoid linting fixes in the future?

File: app/src/main/java/com/cornellappdev/android/eatery/model/CafeModel.java [hunk 9]
```diff
@@ -134,6 +142,51 @@ public void setHours(LocalDate date, List<Interval> hours) {
...
+				}
+				LocalDateTime start = null, end = null;
+				start = LocalTime.parse(event.startTime().toUpperCase().substring(11),
+						timeFormatter).atDate(localDate);
+				end = LocalTime.parse(event.endTime().toUpperCase().substring(11),
+						timeFormatter).atDate(localDate);
+				LocalDateTime midnightTomorrow = localDate.atTime(LocalTime.MIDNIGHT);
+				if (start != null && end != null) {
+					if (end.isBefore(start) && (end.isEqual(midnightTomorrow) || end
+							.isAfter(midnightTomorrow))) {
```
Comment (young-k) [inline app/src/main/java/com/cornellappdev/android/eatery/model/CafeModel.java]: same comment regarding linting, `end` should not be separated from the function call

File: app/src/main/java/com/cornellappdev/android/eatery/model/DiningHallModel.java [hunk 10]
```diff
@@ -133,6 +141,45 @@ public static DiningHallModel fromJSON(Context context, boolean hardcoded, JSONO
...
+						timeFormatter).atDate(localDate);
+				end = LocalTime.parse(event.endTime().toUpperCase().substring(11),
+						timeFormatter).atDate(localDate);
+				MealType type = MealType.fromDescription(event.description());
+				MealModel mealModel = new MealModel(start,end);
+				mealModel.setType(type);
+				MealMenuModel mealMenuModel = MealMenuModel.fromGraphQL(event.menu());
+				mealModel.setMenu(mealMenuModel);
+				if (type != null) {
+					dailyMenuModel.addMeal(type, mealModel);
```
Comment (young-k) [inline app/src/main/java/com/cornellappdev/android/eatery/model/DiningHallModel.java]: i've noticed that indentation is always 8, @ningning621 is this intentional?

File: app/src/main/java/com/cornellappdev/android/eatery/network/JsonUtilities.java [hunk 11]
```diff
@@ -75,4 +79,41 @@ public static String loadJSONFromAsset(Context context, String fileName) {
...
 		return eateryList;
 	}
+
+    public static ArrayList<EateryBaseModel> parseEateries(List<AllEateriesQuery.Eatery> eateries, Context mainContext) {
+        ArrayList<EateryBaseModel> eateryList = new ArrayList<EateryBaseModel>();
+        JSONObject hardCodedData = null;
+        try {
+            hardCodedData = new JSONObject(loadJSONFromAsset(mainContext, "externalEateries.json"));
+            JSONArray hardCodedEateries = hardCodedData.getJSONArray("eateries");
+            for (int i = 0; i < hardCodedEateries.length(); i++) {
```
Comment (young-k) [inline app/src/main/java/com/cornellappdev/android/eatery/network/JsonUtilities.java]: can we not iterate using something like `for (a : b)`?

File: app/src/main/java/com/cornellappdev/android/eatery/network/JsonUtilities.java [hunk 12]
```diff
@@ -75,4 +79,41 @@ public static String loadJSONFromAsset(Context context, String fileName) {
...
+                if (obj.has("id") && DINING_HALL_IDS.contains(obj.getInt("id"))) {
+                    model = DiningHallModel.fromJSON(mainContext, true, obj);
+                } else {
+                    model = CafeModel.fromJSONObject(mainContext, true, obj);
+                }
+                if (model != null) {
+                    eateryList.add(model);
+                }
+            }
+            for (int i = 0; i < eateries.size(); i++) {
```
Comment (young-k) [inline app/src/main/java/com/cornellappdev/android/eatery/network/JsonUtilities.java]: same question here, we're not using `i` in the for loop

File: app/src/main/java/com/cornellappdev/android/eatery/model/DiningHallModel.java (hunk 10 above)
Comment (ningning621) [reply app/src/main/java/com/cornellappdev/android/eatery/model/DiningHallModel.java]: I'll have to look at the codebase. I think we ran a linter last semester and it messed up some of the spacing. I'm pretty certain that the 8 space indentation shows up only on github and not in the IDE.

**Comment by ningning621** [comment]
> Should `.idea/caches/gradle_models.ser` be committed?

No. @2AStudios please remove this file from the commit

**Comment by ningning621** [comment]
> @ningning621 we should also set up linting on PRs for Android code, or at least with a Java linter for now.

The linter that we used last semester messed up some of the indentation, but I'll look for another linter today and run it thru this branch

File: app/src/main/java/com/cornellappdev/android/eatery/network/JsonUtilities.java (hunk 11 above)
Comment (2AStudios) [reply app/src/main/java/com/cornellappdev/android/eatery/network/JsonUtilities.java]: I'll see if I can change this, this was directly taken from the previous logic.

**Comment by 2AStudios** [comment]
> also maybe clean up the tense and "mostly working DataModels" for your commits. @ningning621 idk how nitpicky yall are for commits but we should try to keep the history as clean as possible.

Apologies, I'll read up on what our policy is for commit names next time around. Since commit titles are tied in with the hashing system, I can't retroactively change them as far as I can tell.

File: app/build.gradle (hunk 2 above)
Comment (2AStudios) [reply app/build.gradle]: No clue, but I'll spend some time alphabetizing it anyway.

File: app/src/main/java/com/cornellappdev/android/eatery/MainActivity.java (hunk 5 above)
Comment (2AStudios) [reply app/src/main/java/com/cornellappdev/android/eatery/MainActivity.java]: I see no issues in Android Studio, so I believe this is the linter causing issues.

File: app/src/main/java/com/cornellappdev/android/eatery/network/JsonUtilities.java (hunk 11 above)
Comment (2AStudios) [reply app/src/main/java/com/cornellappdev/android/eatery/network/JsonUtilities.java]: Nope, JSONObject doesn't support iterators as far as I can tell.

**Comment by 2AStudios** [comment]
Reformatted the code and rewrote a few portions for your review.

File: app/src/main/java/com/cornellappdev/android/eatery/MainActivity.java (hunk 3 above)
Comment (2AStudios) [reply app/src/main/java/com/cornellappdev/android/eatery/MainActivity.java]: Done

File: app/src/main/java/com/cornellappdev/android/eatery/network/JsonUtilities.java (hunk 12 above)
Comment (2AStudios) [reply app/src/main/java/com/cornellappdev/android/eatery/network/JsonUtilities.java]: Changed to use an iterator.

File: app/src/main/java/com/cornellappdev/android/eatery/MainActivity.java [hunk 13]
```diff
@@ -14,6 +14,7 @@
 import android.support.v7.widget.LinearLayoutManager;
 import android.support.v7.widget.RecyclerView;
 import android.support.v7.widget.SearchView;
+import android.util.Log;
```
[the remark this answers is not in the document]
Comment (2AStudios) [reply app/src/main/java/com/cornellappdev/android/eatery/MainActivity.java]: Deleted

File: app/src/main/java/com/cornellappdev/android/eatery/model/EateryBaseModel.java [hunk 14]
```diff
@@ -115,6 +117,25 @@ public boolean hasPaymentMethod(PaymentMethod method) {
...
+		mName = eatery.name();
+		mBuildingLocation = eatery.location();
+		mNickName = eatery.nameShort();
+		mLatitude = eatery.coordinates().latitude();
+		mLongitude = eatery.coordinates().longitude();
+
+		List<PaymentMethod> paymentMethods = new ArrayList<>();
+		if(eatery.paymentMethods().brbs())  paymentMethods.add(PaymentMethod.fromShortDescription("debit"));
+        if(eatery.paymentMethods().cash())  paymentMethods.add(PaymentMethod.fromShortDescription("cash"));
+        if(eatery.paymentMethods().cornellCard())paymentMethods.add(PaymentMethod.fromShortDescription("cornellcard"));
```
[the remark this answers is not in the document]
Comment (2AStudios) [reply app/src/main/java/com/cornellappdev/android/eatery/model/EateryBaseModel.java]: At the moment we don't. It doesn't clutter up the code much at all and future proofs it for when we need those options from the object. I'd argue that we don't have to change it.

File: app/src/main/java/com/cornellappdev/android/eatery/network/NetworkUtilities.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (2AStudios) [reply app/src/main/java/com/cornellappdev/android/eatery/network/NetworkUtilities.java]: JSON_FALLBACK logic added, all calls to Log removed.

File: app/build.gradle (hunk 2 above)
Comment (ningning621) [reply app/build.gradle]: I'll do some research and write up a brief Android code style doc for Sunday

**Comment by ningning621** [comment]
@2AStudios Looking good overall! Is it possible to uncommit the `.idea` file? I'll approve this PR after you do so.

File: .idea/gradle.xml [hunk 15]
```diff
@@ -3,6 +3,9 @@
   <component name="GradleSettings">
     <option name="linkedExternalProjectsSettings">
       <GradleProjectSettings>
+        <compositeConfiguration>
```
[the remark this answers is not in the document]
Comment (ningning621) [reply .idea/gradle.xml]: @2AStudios

**Pull request merged**